realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -23,6 +23,79 @@
  * element in the {@code OrderedRealmCollection} has an index. Each element can thus be accessed by its
  * index, with the first index being zero. Normally, {@code OrderedRealmCollection}s allow duplicate
  * elements, as compared to Sets, where elements have to be unique.
+ *
+ * <p>
+ *
+ * There are three types of {@link OrderedRealmCollection}. {@link RealmResults} and {@link RealmList} are live
+ * collections. They are up-to-date all the time and they will never contain a invalid {@link RealmObject}.
+ * {@link OrderedRealmCollectionSnapshot} is different. An {@link OrderedRealmCollectionSnapshot} can be created from
+ * another {@link OrderedRealmCollection}. Its size and elements order stay the same as the original collection's when
+ * it was created. {@link OrderedRealmCollection} may contain invalid {@link RealmObject} if the object gets deleted.
+ *
+ * <p>
+ *
+ * <div class="loops"></div>
+ * Using iterators to iterate on {@link OrderedRealmCollection} will always work. You can delete or modify the elements
+ * without impacting the iterator. See below example:
+ *
+ * <pre>
+ * {@code
+ * RealmResults<Dog> dogs = realm.where(Dog.class).findAll();
+ * int s = dogs.size(); // 10
+ * realm.beginTransaction();
+ * for (Dog dog : dogs) {
+ *     dog.deleteFromRealm();
+ *     s = dogs.size(); // This will be decreased by 1 every time after a dog is removed.
+ * }
+ * realm.commitTransaction();
+ * s = dogs.size(); // 0
+ * }
+ * </pre>
+ *
+ * You can delete the element while iterating without impacting the iterator. But whenever the element in the
+ * {@link RealmResults} changes, the {@link RealmResults} will be impacted. Like the above example, during the iterating
+ * after a element gets deleted, the {@link RealmResults}'s size become different. So does the order if modifications
```

Comment:
I'm confused now. Does this means that if I do

```
RealmResults<Person> persons = realm.where(Person.class).findAllSorted("age", Sort.ASC);
realm.beginTransaction();
for (Person p : persons) {
  p.setAge(persons.max("age") + 1);
}
realm.commitTransaction();
```
That I will keep moving items to the back of the list so they get processed twice?